### ISSUE TITLE
Installation problem:

### ISSUE THREAD

**Issue by fabiobettiol** [NgSekLong/SelectUrCAS#2]
Hi! 
Got this error while installing (Mac M1 - Ventura)
```

docker-compose -f docker-compose.yml -f ./source/authentication/openldap/docker-compose.yml -f ./source/authentication-attribute/openldap/docker-compose.yml -f ./source/cas-client/phpcas/docker-compose.yml -f ./source/ticket-registry/redis/docker-compose.yml -f ./source/service-registry/json-1001/docker-compose.yml -f ./source/helper/nginx-load-balancer/docker-compose.yml up --scale cas_server=2

[+] Building 2.6s (8/8) FINISHED
 => [selecturcas-phpcas internal] load build definition from Dockerfile                                             0.0s
 => => transferring dockerfile: 32B                                                                                 0.0s
 => [selecturcas-cas_server internal] load build definition from Dockerfile                                         0.0s
 => => transferring dockerfile: 32B                                                                                 0.0s
 => [selecturcas-phpcas internal] load .dockerignore                                                                0.0s
 => => transferring context: 2B                                                                                     0.0s
 => [selecturcas-cas_server internal] load .dockerignore                                                            0.0s
 => => transferring context: 34B                                                                                    0.0s
 => CANCELED [selecturcas-phpcas internal] load metadata for docker.io/library/php:7.4-apache                       2.5s
 => ERROR [selecturcas-cas_server internal] load metadata for docker.io/adoptopenjdk/openjdk11:alpine-jre           2.5s
 => [auth] library/php:pull token for registry-1.docker.io                                                          0.0s
 => [auth] adoptopenjdk/openjdk11:pull token for registry-1.docker.io                                               0.0s
------
 > [selecturcas-cas_server internal] load metadata for docker.io/adoptopenjdk/openjdk11:alpine-jre:
------
failed to solve: rpc error: code = Unknown desc = failed to solve with frontend dockerfile.v0: failed to create LLB definition: no match for platform in manifest sha256:f829f93be7762ba6662f7096a542d797091f2030628143080e7429ca7b43009c: not found
```

Thank you!
Fabio

**Comment by NgSekLong** [comment]
Hello Fabio

Thanks for trying out the project!

The project was developed using an Ubuntu 20 machine, I checked just now the command does still works on Ubuntu.

I don't have an Mac on hand to try this out, however there is similar report of this kind of issue indicating a chip issue.
https://github.com/docker/for-mac/issues/5975#issuecomment-942403670

If I get my hand on a Mac (preferable similar spec to `Mac M1 - Ventura`) I will try to fix it, but it might take a while since I don't have one right now.

If you really want to try it out, you can see if you can install a virtual machine, at least a Ubuntu machine should be able to work.

Cheers!
Andy

**Comment by fabiobettiol** [comment]
Hi!

Thank you for your answer.
Yes, I already experienced several problems related to docker and M1 chip
(mostly Apache-related). I’ll proceed  by installing an Ubuntu VM and give
out a try.

I found this tool-and the demo video- very enlightening. As I don’t have
any previous Java or CAS experience, it’s being a real challenge to set it
up CAS. A basic installation is up and running, but there is still a lot to
do.

I’d like to contribute with your efforts, by translating into Spanish any
docs, and even providing subtitles for your videos, if you may consider it
useful.

May be I’d be able to contribute in other ways you may consider it.

Kind regards,
Fabio.

Il giorno ven 2 dic 2022 alle 05:03 NgSekLong ***@***.***> ha
scritto:

**Comment by fabiobettiol** [comment]
Hi! 

Thank you. 
It’s running perfectly on a VM with Ubuntu 20.04

A couple of questions:
- Does the management app is included? Which endpoint ?
- Is it possible for me to view/tweak some CAS options (can.properties, JSON services). How ?

Thank you. 
Best regards,

Fabio

**Comment by NgSekLong** [comment]
> Does the management app is included
Yes, it is on the port `8444`, you need to select it in Other section in the HTML page

> Is it possible for me to view/tweak some CAS options

Yes, I used YAML so you can locate this file https://github.com/NgSekLong/SelectUrCAS/blob/master/source/cas-overlay/cas.yml in your installation and editing it to add in your own customization.

Moreover, you can get inside the docker container using command like this:

1. Execute `docker ps`
2. Copy the docker ID for CAS
3. execute `docker exec -ti xxxx bash`
4. You are in the CAS server docker container

Then you can add your JSON service and other tweaking if you want.

If you want to permently change the setup you would need to implement them using docker-compose, which you might need some study on.

Cheers!
Andy

**Comment by NgSekLong** [comment]
> providing subtitles for your videos

While that would be great, YouTube have disabled the add subtitle functionality :( So I cannot allow you to add even if I want to, bummer...

> I’d like to contribute with your efforts, by translating into Spanish any docs,

I am already greatly appreciate that you tried out the tools! 

Translating into Spanish would be amazing but I think you can do so after you really try it out and think it is worth your time!

I have not update this tool but I did put a lot of time into it beforehand, if you think it is useful when I would consider putting more effort into it. So it is more up to date~

Cheers!
Andy

**Comment by fabiobettiol** [comment]
Thank you, very much.

In a couple of days I have learned a lot using your tool, much more than my frustrating early attempts with CAS.
So … YES!!! I find your tool *very* useful. It would be good for the whole community interested in CAS if you continue the development/update.
 
Thank you! I was able to login into the containers and check the configs.
I have, and apologies for that, some further questions:

1. I can not find how to start Management Application. Sorry, I checked all over the web page and just can’t find it. I started a build with *all* the options active. Could not find a container listening in 8444
2. Is there a way to convert the YAML config files to the “text” syntax (i.e. “cas.properties") as in the CAS documentation?
3. Would you consider offering some kind of paid classes for teaching CAS to others.? I’d be interested.

Thank you. I really appreciate not only for your tool, but also for the time you spend on this kind answers.

Best regards,
Fabio
